torch/jit/__init__.py

```diff
@@ -1304,7 +1317,10 @@ def check_unique(param):
             raise ValueError("Modules that have hooks assigned can't be compiled")
 
         for name, submodule in orig._modules.items():
-            self._modules[name] = TracedModule(submodule, id_set, optimize=optimize)
+            if isinstance(submodule, ScriptModule) and not isinstance(submodule, TracedModule):
+                self._modules[name] = submodule.copy()
```

Comment:
This goes around our parameter uniqueness checks, and it seems unsafe to me. Are you sure this is correct?